uwblueprint/project-read-backend
# create data import script

### Notion ticket link


[Data import](https://www.notion.so/uwblueprintexecs/Data-import-b41b697042c54518b788369640fb28b7)

### Implementation description

- Wrote a script that imports students in the registration form: `python manage.py load-registration [REGISTRATION_FORM].csv [FIELDS_MAP].json`

### Steps to test

1. Download registration form as CSV
2. Create a map json that maps default fields to its corresponding column header name in the csv for parents/family. For children, we must map both the default and dynamic fields. An example is shown below:
```json
{
  "email": "email",
  "home_number": "home phone",
  "cell_number": "cell phone",
  "work_number": "alternative phone",
  "address": "Address",
  "first_name": "First name",
  "last_name": "Last name",
  "date_of_birth": "parent DOB - don't have to ask on the phone",
  "children": [
    {
      "default_fields": {
        "first_name": "child 1 first name",
        "last_name": "child 1 last name"
      },
      "dynamic_fields": {
        "allergies": "child 1 allergies",
        "gender": "child 1 gender",
        "date of birth (old)": "child 1 DOB"
      }
    },
    {
      "default_fields": {
        "first_name": "child 2 first name",
        "last_name": "child 2 last name"
      },
      "dynamic_fields": {
        "allergies": "child 2 allergies",
        "gender": "child 2 gender",
        "date of birth (old)": "child 2 DOB"
      }
    },
    {
      "default_fields": {
        "first_name": "child 3 first name",
        "last_name": "child 3 last name"
      },
      "dynamic_fields": {
        "allergies": "child 3 allergies",
        "gender": "child 3 gender",
        "date of birth (old)": "child 3 DOB"
      }
    }
  ]
}

```
4. Run the script
5. Check that families, parents, children, and dynamic fields were all correctly imported

### What should reviewers focus on?

- Are there any edge cases from the registration form that would break this script?
